src/Gui/MainWindow.cpp

```diff
@@ -150,11 +150,91 @@ class CustomMessageEvent : public QEvent
     int _timeout;
 };
 
+class DimensionWidget : public QPushButton, WindowParameter
+{
+    Q_OBJECT
+
+public:
+    DimensionWidget(QWidget* parent): QPushButton(parent), WindowParameter("Units")
+    {
+        setFlat(true);
+        setText(qApp->translate("Gui::MainWindow", "Dimension"));
+        setMinimumWidth(120);
+
+        //create the action buttons
+        auto* menu = new QMenu(this);
+        auto* actionGrp = new QActionGroup(menu);
+        int num = static_cast<int>(Base::UnitSystem::NumUnitSystemTypes);
+        for (int i = 0; i < num; i++) {
```

Comment:
Maybe https://github.com/FreeCAD/FreeCAD/pull/8870#discussion_r1140461629

Reply:
I thought to that too but that didn't really lighten the code actually.
I'm not sure to fully understand your comment there though as QActionGroup has no signal, so in any case we have to connect to each individual QAction signal.

Reply:
> QActionGroup has no signal

It actually has two: hovered(QAction*) and triggered(QAction*)
It looks like the Qt docs for Qt6 isn't complete because they are not mentioned while the Qt5 docs show them. And both signals are declared in the QActionGroup for Qt5 and Qt6, so they definitely exist.

Reply:
Yep I was fooled by Qt6 doc.
I made the change in a last commit if you to check it matches your expectations.

Reply:
I will have a look in one or two days as I am still working on the image plane/view migration to the core system.

Reply:
OK, I will polish the PR and then let you merge if it's fine. ;)

Reply:
@wwmayer May you tell when you'll review ? PR is ready.
If you have no time for this soon, I think I'll merge as is and then do another PR to take into account your further remarks if needed. Thx

Reply:
> If you have no time for this soon, I think I'll merge as is and then do another PR to take into account your further remarks if needed.

The code looks good and using the QActionGroup this way is exactly what I had in mind. So, I think the code can be merged now.